client/src/components/modules/MapCard.js

```diff
@@ -6,94 +6,104 @@ import {
   DashboardTwoTone,
   DownloadOutlined,
 } from "@ant-design/icons";
-import { Link } from "@reach/router";
 import CustomMapBadge from "../../public/custom-map-badge.svg";
 import CustomSongBadge from "../../public/custom-song-badge.svg";
 
 import { Card, Popconfirm, Tooltip } from "antd";
 import "./MapCard.css";
 import DefaultBG from "../../public/default-bg.png";
 
-class MapCard extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
-  }
-
-  render() {
-    return (
-      <Card
-        title={
-          <div className="MapCard-title">
-            <div className={`MapCard-icon mod-${this.props.mod}`}></div>
-            {`${this.props.mod}${this.props.index}`}
-            <div style={{ marginRight: 12 }} />
-            {this.props.customMap && (
-              <Tooltip title="GTS Custom Map">
-                <img className="MapCard-overlay-icon" src={CustomMapBadge} />
-              </Tooltip>
-            )}
-            {this.props.customSong && (
-              <Tooltip title="GTS Custom Song">
-                <img className="MapCard-overlay-icon" src={CustomSongBadge} />
-              </Tooltip>
-            )}
-          </div>
-        }
-        bordered={true}
-        cover={
-          <a target="_blank" href={`https://osu.ppy.sh/b/${this.props.mapId}`} style={{ position: "relative" }}>
-            <img src={this.props.image} onError={(e) => (e.target.src = DefaultBG)} />
+export default function MapCard({
+  _id,
+  artist,
+  bpm,
+  creator,
+  customMap,
+  customSong,
+  diff,
+  handleDelete,
+  hp,
+  image,
+  index,
+  isPooler,
+  length,
+  mapId,
+  mod,
+  od,
+  pooler,
+  sr,
+  title,
```

Comment:
it would probably be a good idea to just have `map` as a prop instead of unpacking all the fields and creating a huge list
but that doesn't have to be done in this pr, i can do that later